### PR TITLE
Update envoy to e2117cb

### PR DESCRIPTION
**Pull request by lizan**
**What this PR does / why we need it**:
Update Envoy to https://github.com/envoyproxy/envoy/commit/e2117cbf1f7a84e4cfd7a1e00bcc053b1afb0ea7

Align with https://github.com/envoyproxy/envoy/pull/2386

**Release note**:
```release-note
None
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -35,7 +35,7 @@ bind(
 )
 
 # When updating envoy sha manually please update the sha in istio.deps file also
```
Comment (rkpagadala) [inline WORKSPACE]: Lizan,
   Please update istio.deps also at the same time.

Comment (lizan) [reply WORKSPACE]: Done

**Comment by lizan** [comment]
TSAN failure is likely due to #1124, temporary disable Google gRPC to pass presubmit here, and make another fix later.

File: prow/proxy-presubmit.sh [hunk 2]
```diff
@@ -45,6 +45,8 @@ fi
 
 cd $ROOT
 
+export BAZEL_TEST_ARGS="--test_output=errors --define=google_grpc=disabled"
```
Comment (qiwzhang) [inline prow/proxy-presubmit.sh]: My concerns is:  test is using different code than release.  This is not testing anything.

Comment (qiwzhang) [reply prow/proxy-presubmit.sh]: Just realized this is for "auth",  auth is not using grpc so it is fine then.

**Comment by lizan** [comment]
in mixer we are not using Google gRPC either.

**Pull request merged**